sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriter.java

```diff
@@ -196,8 +205,9 @@ Mono<StoreResponse> writePrivateAsync(
                 } catch (Exception e) {
                     return Mono.error(e);
                 }
-
-                replicaStatusList.set(Arrays.asList(primaryUri.getHealthStatusDiagnosticString()));
+                replicaStatuses.remove(primaryUri);
+                replicaStatusList.put(Uri.NOT_ATTEMPTING, new ArrayList<>(replicaStatuses.values()));
```

Comment:
Is the new List and copying the values necessary?

Reply:
I was doing it convert into a list because .values() returns a collection. Decided to change it to a set instead of using a map and a list. No copying over is happening now.